addons/pos_restaurant/static/src/app/services/pos_store.js

```diff
@@ -46,6 +48,54 @@ patch(PosStore.prototype, {
         }
         return super.defaultScreen;
     },
+    createNewOrder(data) {
+        const order = super.createNewOrder(data);
+
+        if (order.table_id) {
+            order.setCustomerCount(order.table_id.seats);
+        }
+
+        return order;
+    },
+    async setCustomerCount(o = false) {
+        const currentOrder = o || this.getOrder();
+        const count = await makeAwaitable(this.dialog, NumberPopup, {
+            feedback: (buffer) => {
+                const value = this.env.utils.formatCurrency(
+                    currentOrder?.amountPerGuest(parseInt(buffer, 10) || 0) || 0
+                );
+                return value ? `${value} / ${_t("Guest")}` : "";
+            },
+        });
+        const guestCount = parseInt(count, 10) || 0;
+        if (guestCount == 0 && currentOrder.lines.length === 0) {
+            this.removeOrder(currentOrder);
+            this.showScreen("FloorScreen");
+            return false;
+        }
+        currentOrder.setCustomerCount(guestCount);
+        this.addPendingOrder([currentOrder.id]);
+        return true;
+    },
```

Comment:
```suggestion
    async setCustomerCount(order) {
        const count = await makeAwaitable(this.dialog, NumberPopup, {
            feedback: (buffer) => {
                const value = this.env.utils.formatCurrency(
                    order.amountPerGuest(parseInt(buffer, 10) || 0) || 0
                );
                return value ? `${value} / ${_t("Guest")}` : "";
            },
        });
        const guestCount = parseInt(count, 10) || 0;
        if (guestCount == 0 && order.lines.length === 0) {
            this.removeOrder(order);
            this.showScreen("FloorScreen");
            return false;
        }
        order.setCustomerCount(guestCount);
        this.addPendingOrder([order.id]);
        return true;
    },
```

Define this function such that `order` input is required. Besides, you're already assuming that that's the case in line 71 (`currentOrder.lines.length`).